### PR TITLE
UI, Slate: rename ShowSignal to EngageSignal, implement ReplaceSignal

### PR DESCRIPTION
**Pull request by nhaagen**
Contents of Slates may be replaced asynchronously.

### CODE REVIEW

File: src/UI/Implementation/Component/MainControls/Slate/Slate.php [hunk 1]
```diff
@@ -66,7 +72,8 @@ public function __construct(
     protected function initSignals()
     {
         $this->toggle_signal = $this->signal_generator->create();
-        $this->show_signal = $this->signal_generator->create();
+        $this->engage_signal = $this->signal_generator->create();
+        $this->replace_signal = $this->signal_generator->create("ILIAS\\UI\\Implementation\\Component\\ReplaceSignal");
```
Comment (klees) [inline src/UI/Implementation/Component/MainControls/Slate/Slate.php]:
```suggestion
        $this->replace_signal = $this->signal_generator->create(ILIAS\\UI\\Implementation\\Component\\ReplaceSignal::class);
```
This is not just any string, but a class name.

File: src/UI/Implementation/Component/MainControls/Slate/Renderer.php [hunk 2]
```diff
@@ -66,19 +66,17 @@ protected function renderSlate(
...
+            'toggle' => $component->getToggleSignal(),
+            'engage' => $component->getEngageSignal(),
+            'replace' => $component->getReplaceSignal()
+        ];
+        $component = $component->withAdditionalOnLoadCode(function ($id) use ($slate_signals) {
+            $js = "fn = il.UI.maincontrols.slate.onSignal;";
+            foreach ($slate_signals as $key => $signal) {
+                $js .= "$(document).on('{$signal}', function(event, signalData) { fn('{$key}', event, signalData, '{$id}'); return false;});";
+            }
+            return $js;
```
Comment (klees) [inline src/UI/Implementation/Component/MainControls/Slate/Renderer.php:79]: :+1: Thanks for boyscouting.

**Review by Amstutz — APPROVED** [review]

**Comment by matthiaskunkel** [comment]
Jour Fixe, 21 OCT 2019 : We highly appreciate this suggestion and accept the PR for trunk. Please feel free to merge to trunk.

**Comment by Amstutz** [comment]
Thx a lot!

**Pull request merged**